### PR TITLE
Adopt @huebert's Pythonic base class into Pytree.

### PR DESCRIPTION
**Pull request by littleredcomputer**
add a unit test for Pytree.partial with scan (see GEN-339)

### CODE REVIEW

File: src/genjax/_src/core/pytree.py [hunk 1]
```diff
@@ -494,3 +495,35 @@ def f(v, x):
 
     def __call__(self, *args, **kwargs):
         return self.fn(*self.dyn_args, *args, **kwargs)
+
+
+def nth(x: Pytree, idx: Int):
+    return jtu.tree_map(lambda v: v[idx], x)
+
+
+class Pythonic(Pytree):
```
Comment (littleredcomputer) [inline src/genjax/_src/core/pytree.py]: Maybe the name should be PythonicPytree

Comment (littleredcomputer) [reply src/genjax/_src/core/pytree.py]: I tried making this a decorator but learned for myself what had stopped @mhuebert from doing so: we would need a type incantation that says "This function accepts a _class_ K and returns a class K' with additional member functions." That sounds an awful lot like saying "Class Pythonic extends Pytree in the following way," and a base class is the least tricky way of accomplishing this.

File: src/genjax/_src/core/pytree.py [hunk 2]
```diff
@@ -494,3 +495,35 @@ def f(v, x):
 
     def __call__(self, *args, **kwargs):
         return self.fn(*self.dyn_args, *args, **kwargs)
+
+
+def nth(x: Pytree, idx: Int):
```
Comment (sritchie) [inline src/genjax/_src/core/pytree.py:500]: can we add docstrings to these, noting what they do for pytree inheritors? can be short, even LLM-generated if you're on Cursor!

File: src/genjax/_src/core/pytree.py (hunk 1 above)
Comment (sritchie) [reply src/genjax/_src/core/pytree.py]: that's fine! I like that name `PythonicPytree` as well.

File: tests/generative_functions/test_scan_combinator.py [hunk 3]
```diff
@@ -338,3 +338,48 @@ def model(k):
...
         assert w < -100.0
+
+
+class TestScanWithParameters:
+    @pytest.fixture
+    def key(self):
+        return jax.random.PRNGKey(314159)
+
+    @staticmethod
+    def gen_partial(gen_fn, closed_over):
```
Comment (sritchie) [inline tests/generative_functions/test_scan_combinator.py]: I don't see this used anywhere...

File: src/genjax/_src/core/pytree.py (hunk 1 above)
Comment (littleredcomputer) [reply src/genjax/_src/core/pytree.py]: done

File: src/genjax/_src/core/pytree.py (hunk 2 above)
Comment (littleredcomputer) [reply src/genjax/_src/core/pytree.py:500]: done

**Review by mhuebert — APPROVED** [review]
LGTM!

**Pull request merged**